rts/motoko-rts/src/region.rs

```diff
@@ -948,6 +948,32 @@ pub(crate) unsafe fn region_load<M: Memory>(_mem: &mut M, r: Value, offset: u64,
     }
 }
 
+pub unsafe fn region_write_heap_shapshot<M: Memory>(mem: &mut M, r: Value) {
```

Comment:
This looks very good. If the the Rust function `from_raw_parts` would not artificially limit the size to `isize::MAX`, it would be even simpler...

Reply:
💯